src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -304,4 +305,25 @@ private static function computeRegexp(array $tokens, int $index, int $firstOptio
             }
         }
     }
+
+    private static function removeCapturingGroups(string $regexp)
+    {
+        for ($i = 0; $i < \strlen($regexp); ++$i) {
+            if ('\\' === $regexp[$i]) {
+                ++$i;
+                continue;
+            }
+            if ('(' !== $regexp[$i] || !isset($regexp[$i + 2])) {
```

Comment:
What if there is an escaped `(` in the regex?

Reply:
This is covered, see code and tests

Reply:
Ah yes indeed. Missed that.